src/v/utils/container_utils.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/coroutine/maybe_yield.hh>
+
+template<typename Container>
+concept IterableContainer
+  = requires(Container c, typename Container::iterator i) {
+    { c.begin() } -> std::same_as<typename Container::iterator>;
+    { c.end() } -> std::same_as<typename Container::iterator>;
+    { ++i } -> std::convertible_to<typename Container::iterator>;
+};
+
+template<typename Container>
+concept RangeErasableContainer = requires(
+  Container c, typename Container::iterator i, typename Container::iterator j) {
+    { c.erase(i, j) } -> std::same_as<typename Container::iterator>;
+};
+
+template<typename Container>
+concept SizableContainer = requires(Container c) {
```

Comment:
I think we should have some naming conventions, the standard uses `snake_case`, such as `std::forward_range` and `std::sized_range` (note, I'm not yet proposing we bring in `<range>`.

But I do think we should be careful. So I suggest:
* `snake_case`
* use namespaces, at least until we have a more refined idea of what concepts we want.
* This one could be `has_member_size`

Reply:
I went with `PascalCase` since most template parameters are written that way as well. That being said I'm probably going to remove all concepts from this PR since I'm no longer using them in the function. It would be nice to have a standardize utility library of concepts though for folks to use.